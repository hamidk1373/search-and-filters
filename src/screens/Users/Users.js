/* eslint-disable react-hooks/exhaustive-deps */
import React from "react";
import ProfileCard from "../../components/ProfileCard";
import { useStyles } from "./index.style";
import { Paper } from "@material-ui/core";
import ManIcon from "../../assets/images/man.svg";
import WomanIcon from "../../assets/images/woman.svg";

export default function Users({
  infoInputValue,
  handleChangeSearchInput,
  selectedGender,
  setSelectedGender,
  filteredProfilesList,
}) {
  const classes = useStyles();

  return (
    <>
      <Paper className={classes.filterBox}>
        <div className={classes.inputSearchBox}>
          <label className={classes.searchLabel}>First Name</label>
          <input
            value={infoInputValue}
            onChange={handleChangeSearchInput}
            className={classes.searchInput}
            placeholder='Search in first names....'
          />
        </div>
        <div className={classes.genderSelectorBox}>
          <img
            src={ManIcon}
            alt='Man'
            className={[
              classes.genderIcon,
              selectedGender === "Male" ? classes.selectedGenderIcon : "",
            ].join(" ")}
            onClick={() => setSelectedGender("Male")}
          />
          <img
            src={WomanIcon}
            alt='Woman'
            className={[
              classes.genderIcon,
              selectedGender === "Female" ? classes.selectedGenderIcon : "",
            ].join(" ")}
            onClick={() => setSelectedGender("Female")}
          />
          <div onClick={() => setSelectedGender("")}>X</div>
        </div>
      </Paper>
      <div data-testid='userslist' className={classes.usersList}>
        {filteredProfilesList.map((item) => (
          <ProfileCard key={item.id} data={item} />
        ))}
      </div>
    </>
  );
}

// (x) => {
//
//  reutrn <div></div>
// }

// (x) => <div></div>
