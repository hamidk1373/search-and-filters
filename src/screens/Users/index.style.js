import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
  usersList: {
    display: "grid",
    gridTemplateColumns: "repeat(5, 1fr)",
    gap: "0.5rem",
    padding: "3rem 10rem",
  },
  filterBox: {
    padding: "1rem",
    margin: "2rem 10rem 0",
    display: "flex",
  },
  inputSearchBox: {
    flex: 2,
    border: "1px solid black",
    padding: "0.5rem",
    borderTopLeftRadius: "0.5rem",
    borderBottomLeftRadius: "0.5rem",
    position: "relative",
  },
  genderSelectorBox: {
    flex: 1,
    border: "1px solid black",
    borderTopRightRadius: "0.5rem",
    borderBottomRightRadius: "0.5rem",
    display: "flex",
    justifyContent: "space-around",
    alignItems: "center",
  },
  searchInput: {
    width: "90%",
    padding: "0.5rem",
    borderRadius: "0.5rem",
    border: "none",
    "&:focus": {
      outline: "none",
    },
  },
  searchLabel: {
    position: "absolute",
    backgroundColor: "white",
    top: -11,
    left: "0.7rem",
    paddingRight: 6,
    paddingLeft: 2,
  },
  genderIcon: {
    width: 40,
    cursor: "pointer",
  },
  selectedGenderIcon: {
    border: "1px solid green",
    borderRadius: "0.5rem",
  },
}));
