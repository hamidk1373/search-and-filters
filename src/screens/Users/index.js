/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from "react";
import { profilesMock } from "../../mockData/profilesMock";
import Users from "./Users";

export default function Index() {
  // eslint-disable-next-line no-unused-vars
  const [profilesList, setProfilesList] = useState(profilesMock);
  const [filteredProfilesList, setFilteredProfilesList] = useState(
    profilesMock
  );

  const [infoInputValue, setInfoInputValue] = useState("");
  const [selectedGender, setSelectedGender] = useState("");

  const handleChangeSearchInput = (event) => {
    setInfoInputValue(event.target.value);
  };

  useEffect(() => {
    let newFilteredList = profilesList;

    if (infoInputValue.length > 2) {
      newFilteredList = newFilteredList.filter((item) =>
        item.first_name.toLowerCase().includes(infoInputValue.toLowerCase())
      );
    }

    if (selectedGender) {
      newFilteredList = newFilteredList.filter(
        (item) => item.gender === selectedGender
      );
    }

    setFilteredProfilesList([...newFilteredList]);
  }, [selectedGender, infoInputValue]);

  return (
    <Users
      {...{
        infoInputValue,
        handleChangeSearchInput,
        selectedGender,
        setSelectedGender,
        filteredProfilesList,
      }}
    />
  );
}

// (x) => {
//
//  reutrn <div></div>
// }

// (x) => <div></div>
