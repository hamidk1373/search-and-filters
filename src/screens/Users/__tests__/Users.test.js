import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { create } from "react-test-renderer";
import Users from "../index";

test("Users renders correctly", () => {
  const root = create(
    <BrowserRouter>
      <Users />
    </BrowserRouter>
  );

  expect(root.toJSON()).toMatchSnapshot();
});

test("Users renders correctly", () => {
  const { container, getByText } = render(
    <BrowserRouter>
      <Users />
    </BrowserRouter>
  );

  container.querySelector('[data-testid="userslist"]');

  expect(container).toBeInTheDocument();

  console.log("container.children", container.children[0]);

  expect(container.children.length).toBe(30);
});
