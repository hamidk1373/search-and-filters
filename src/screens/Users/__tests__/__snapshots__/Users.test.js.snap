// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders correctly 1`] = `
Array [
  <div
    className="MuiPaper-root makeStyles-filterBox-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="makeStyles-inputSearchBox-3"
    >
      <label
        className="makeStyles-searchLabel-6"
      >
        First Name
      </label>
      <input
        className="makeStyles-searchInput-5"
        onChange={[Function]}
        placeholder="Search in first names...."
        value=""
      />
    </div>
    <div
      className="makeStyles-genderSelectorBox-4"
    >
      <img
        alt="Man"
        className="makeStyles-genderIcon-7 "
        onClick={[Function]}
        src="man.svg"
      />
      <img
        alt="Woman"
        className="makeStyles-genderIcon-7 "
        onClick={[Function]}
        src="woman.svg"
      />
      <div
        onClick={[Function]}
      >
        X
      </div>
    </div>
  </div>,
  <div
    className="makeStyles-usersList-1"
    data-testid="userslist"
  >
    <a
      href="/users/1"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Emogene
      </div>
    </a>
    <a
      href="/users/2"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Wye
      </div>
    </a>
    <a
      href="/users/3"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Chandler
      </div>
    </a>
    <a
      href="/users/4"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Damiano
      </div>
    </a>
    <a
      href="/users/5"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Amory
      </div>
    </a>
    <a
      href="/users/6"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Chip
      </div>
    </a>
    <a
      href="/users/7"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Alexina
      </div>
    </a>
    <a
      href="/users/8"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Kayne
      </div>
    </a>
    <a
      href="/users/9"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Noellyn
      </div>
    </a>
    <a
      href="/users/10"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Ewell
      </div>
    </a>
    <a
      href="/users/11"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Bessie
      </div>
    </a>
    <a
      href="/users/12"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Charity
      </div>
    </a>
    <a
      href="/users/13"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Ole
      </div>
    </a>
    <a
      href="/users/14"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Wolf
      </div>
    </a>
    <a
      href="/users/15"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Bridie
      </div>
    </a>
    <a
      href="/users/16"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Estella
      </div>
    </a>
    <a
      href="/users/17"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Chet
      </div>
    </a>
    <a
      href="/users/18"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Marlowe
      </div>
    </a>
    <a
      href="/users/19"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Glynda
      </div>
    </a>
    <a
      href="/users/20"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Barbaraanne
      </div>
    </a>
    <a
      href="/users/21"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Kristan
      </div>
    </a>
    <a
      href="/users/22"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Lisbeth
      </div>
    </a>
    <a
      href="/users/23"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Welch
      </div>
    </a>
    <a
      href="/users/24"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Upton
      </div>
    </a>
    <a
      href="/users/25"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Alvinia
      </div>
    </a>
    <a
      href="/users/26"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Owen
      </div>
    </a>
    <a
      href="/users/27"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Miles
      </div>
    </a>
    <a
      href="/users/28"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Dennie
      </div>
    </a>
    <a
      href="/users/29"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Mitchell
      </div>
    </a>
    <a
      href="/users/30"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        Aretha
      </div>
    </a>
  </div>,
]
`;
